src/domain/usecases/ExportUsersUseCase.ts

```diff
@@ -0,0 +1,153 @@
+                );
+                const header = columns.map(this.getColumnNameFromProperty);
+                const table = [header, ...userRows];
+                return Papa.unparse(table);
+            }
+        }
+    }
+
+    private getColumnNameFromProperty(property: ColumnMappingKeys): string {
+        return columnNameFromPropertyMapping[property] || property;
+    }
+
+    private formatDate(stringDate?: Date | null): string | undefined {
+        return stringDate ? moment(stringDate).format("YYYY-MM-DD HH:mm:ss") : undefined;
+    }
+
+    private namesFromCollection(
+        collection: User[
+            | "userRoles"
+            | "userGroups"
+            | "organisationUnits"
+            | "dataViewOrganisationUnits"
+            | "searchOrganisationsUnits"],
+        field: OrgUnitKey | typeof defaultNameField,
+        toString: boolean
+    ): string | string[] {
+        // Check if field is in the collection, fallback on default name field
+        const nameField = _.some(collection, field) ? field : defaultNameField;
+        const namesArray = _(collection).map(nameField).value();
+
+        return toString ? namesArray.join(fieldSplitChar) : namesArray;
+    }
+
+    private getPlainUser(
+        user: User,
+        columns: ColumnMappingKeys[],
+        orgUnitsField: OrgUnitKey,
+        toString: boolean
+    ): Record<ColumnMappingKeys, typeof user[ColumnMappingKeys] | string[]> {
+        return _.pick(
+            {
+                ...user,
+                lastUpdated: this.formatDate(user.lastUpdated),
+                lastLogin: this.formatDate(user.lastLogin),
+                created: this.formatDate(user.created),
+                userRoles: this.namesFromCollection(user.userRoles, defaultNameField, toString),
+                userGroups: this.namesFromCollection(user.userGroups, defaultNameField, toString),
+                organisationUnits: this.namesFromCollection(user.organisationUnits, orgUnitsField, toString),
+                dataViewOrganisationUnits: this.namesFromCollection(
+                    user.dataViewOrganisationUnits,
+                    orgUnitsField,
+                    toString
+                ),
+                searchOrganisationsUnits: this.namesFromCollection(
+                    user.searchOrganisationsUnits,
+                    orgUnitsField,
+                    toString
+                ),
+                createdBy: user.createdBy?.username,
+                lastModifiedBy: user.lastModifiedBy?.username,
+            },
+            columns
+        );
+    }
+}
+
+export type ColumnMappingKeys = keyof typeof columnNameFromPropertyMapping;
+
+export type AllowedExportFormat = "json" | "csv";
+
+export type ExportUsersUseCaseOptions = {
+    name: string;
+    columns: ColumnMappingKeys[];
+    filterOptions?: ListOptions;
+    format: AllowedExportFormat;
+    orgUnitsField: OrgUnitKey;
+    isEmptyTemplate?: boolean;
```

Comment:
For this case what I mean is to remove the optional parameter so it should be `isEmptyTemplate: boolean` (without the `?`). We try to avoid this kind of parameters because is easy to forget sending the value.

Example:

```ts
function test(options: OptionsType) {
    console.log(options.isEmpty)
}
```
```ts
type OptionsType = { name: string; isEmpty: boolean; }

test({ name: "", isEmpty: true }) // pass
test({ name: "", })  // error
```

with optional parameter

```ts
test({ name: "", isEmpty: true }) // pass
test({ name: "", })  // pass. 
// Not a big deal for a boolean since undefined/false evaluate to false, 
// but as a convention we try to avoid it for every case.
```

Does it make sense?

Reply:
Yes I understand it maybe be more explicit 👍